### PR TITLE
Displayed error message for already used channel name

### PR DESCRIPTION
**Pull request by harshilsharma63**
#### Summary
In create channel modal, fixed regression bug where user is not displayed error if a channel name (not display name) is already used in team.
In GM to private channel conversion, displayed the same error message if user tries to use a channel name that's already being used in selected team.

#### Ticket Link
Fixes https://mattermost.atlassian.net/browse/MM-54713

#### Release Note

```release-note
None
```

### CODE REVIEW

**Comment by hanzei** [comment]
Out of curiosity: Did you found out what caused the regression?

**Review by larkox — APPROVED** [review]
Nothing blocking, but a concern on how it is solved on the server code.

**Comment by harshilsharma63** [comment]
> Out of curiosity: Did you find out what caused the regression?

Yes. I moved the channel name and URL stuff into a new component and wasn't passing it API response errors.

**Comment by hanzei** [comment]
Thanks for looking into the root cause @harshilsharma63. Does it make sense to add an E2E test in this case as the refactor already caused a regression?

File: server/channels/app/channel.go [hunk 1]
```diff
@@ -3535,6 +3535,20 @@ func (a *App) ConvertGroupMessageToChannel(c request.CTX, convertedByUserId stri
...
+				if errByName != nil {
+					return nil, errByName
+				}
+
+				if duplicateChannel != nil {
+					return nil, model.NewAppError("ConvertGroupMessageToChannel", store.ChannelExistsError, nil, "", http.StatusBadRequest).Wrap(appErr)
+				}
+			}
+		}
+
```
[the remark this answers is not in the document]
Comment (harshilsharma63) [reply server/channels/app/channel.go]: Done.

**Comment by harshilsharma63** [comment]
> Thanks for looking into the root cause @harshilsharma63. Does it make sense to add an E2E test in this case as the refactor already caused a regression?

Yes.

**Review by M-ZubairAhmed — APPROVED** [review]

**Comment by harshilsharma63** [comment]
> Thanks for looking into the root cause @harshilsharma63. Does it make sense to add an E2E test in this case as the refactor already caused a regression?

@hanzei there is actually already cypress test for catching this - `existing_channel_name_spec.js`. It does fail on master, and pass on this branch. Why it didn't got reported in the process, i'm not sure.

**Comment by hanzei** [comment]
cc @saturninoabril on the conversation regarding the E2E test

**Pull request merged**